src/core-layers/path-layer/path-layer-vertex.glsl.js

```diff
@@ -122,49 +113,74 @@ vec3 lineJoin(vec3 prevPoint, vec3 currPoint, vec3 nextPoint) {
     ),
     step(0.0, cornerPosition)
   );
-  offsetVec = mix(miterVec, -tangent, step(0.5, cornerPosition));
+
+  vec2 offsetVec = mix(miterVec, -tangent, step(0.5, cornerPosition));
   offsetScale = mix(offsetScale, 1.0 / max(cosHalfA, 0.001), step(0.5, cornerPosition));
 
   // special treatment for start cap and end cap
-  float isStartCap = step(0.0, -lenA);
-  float isEndCap = step(0.0, -lenB);
-  float isCap = max(isStartCap, isEndCap);
+  bool isStartCap = step(0.0, -lenA) > 0.5;
```

Comment:
This has an issue. len is always positive because it is length.
Step returns zero if -lenA<0, so practically this is a comparison of lenA with zero, with lots of problems because of the -lenA. Can we use EPSILON?